### PR TITLE
Fix resampling of fsnative structural surfaces to fsLR

### PR DESCRIPTION
**Pull request by madisoth**
## Changes proposed in this pull request

- add missing affine/warpfield xfm steps to 32k-resampled surfaces
- replace incorrect std sphere used in concatenation of xfms to fsLR
- make some node names more descriptive (e.g. specify native or 32k surface res, add "resample" to node that performing resampling)
- change incorrect "MNI152Lin" in volume output filenames to "MNI152NLin"  
- fix inconsistent use of "lh" / "rh" vs. "left" / "right" in node names (instances of "left" / "right" have been changed to "lh"/"rh"

### CODE REVIEW

File: xcp_d/workflow/anatomical.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
-# vi: set ft=python sts=4 ts=4 sw=4 et:
+# vi: set ft=python sts=4 ts=4 sw=4 init_structral_wfbet:
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:2]: Is this a typo?

Comment (madisoth) [reply xcp_d/workflow/anatomical.py:2]: yep, thanks for catching -- will fix that right now

File: xcp_d/workflow/anatomical.py [hunk 2]
```diff
@@ -31,6 +32,7 @@
     SurfaceSphereProjectUnproject,
 )
 from xcp_d.utils.bids import collect_data
+from xcp_d.utils.ciftiresample import CiftiSurfaceResample
 from xcp_d.utils.concantenation import _getsesid
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:36]:
```suggestion
from xcp_d.utils.concantenation import _getsesid
```

File: xcp_d/workflow/anatomical.py [hunk 3]
```diff
@@ -953,134 +988,234 @@ def init_anatomical_wf(
...
+                        lh_native_apply_affine,
+                        lh_native_apply_warpfield,
                         [("out_file", "in_file")],
                     ),
                     (
                         remerge_warpfield,
-                        lh_surface_apply_warpfield,
-                        [("out_file", "forward_warp")],
+                        lh_native_apply_warpfield,
+                        [("merged_file", "forward_warp")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1002]:
```suggestion
                        [("out_file", "forward_warp")],
```

File: xcp_d/workflow/anatomical.py [hunk 4]
```diff
@@ -953,134 +988,234 @@ def init_anatomical_wf(
...
                     ),
                     (
                         remerge_inv_warpfield,
-                        lh_surface_apply_warpfield,
-                        [("out_file", "warpfield")],
+                        lh_native_apply_warpfield,
+                        [("merged_file", "warpfield")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1007]:
```suggestion
                        [("out_file", "warpfield")],
```

File: xcp_d/workflow/anatomical.py [hunk 5]
```diff
@@ -953,134 +988,234 @@ def init_anatomical_wf(
...
-                        lh_32k_wm_wf,
-                        [("out_file", "current_sphere")],
+                        convert_xfm2world,
+                        lh_32k_apply_affine,
+                        [("out_file", "affine")],
+                    ),
+                    (
+                        remerge_warpfield,
+                        lh_32k_apply_warpfield,
+                        [("merged_file", "forward_warp")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1032]:
```suggestion
                        [("out_file", "forward_warp")],

```

File: xcp_d/workflow/anatomical.py [hunk 6]
```diff
@@ -953,134 +988,234 @@ def init_anatomical_wf(
...
                     ),
-                    (lh_32k_midthick_wf, ds_midLsurf_wf, [("out_file", "in_file")]),
-                    (lh_32k_pial_wf, ds_pialLsurf_wf, [("out_file", "in_file")]),
-                    (lh_32k_wm_wf, ds_wmLsurf_wf, [("out_file", "in_file")]),
+                    (
+                        remerge_inv_warpfield,
+                        lh_32k_apply_warpfield,
+                        [("merged_file", "warpfield")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1037]:
```suggestion
                        [("out_file", "warpfield")],

```

File: xcp_d/workflow/anatomical.py [hunk 7]
```diff
@@ -953,134 +988,234 @@ def init_anatomical_wf(
...
-                        [("out_file", "forward_warp")],
+                        rh_native_apply_warpfield,
+                        [("merged_file", "forward_warp")],
                     ),
                     (
                         remerge_inv_warpfield,
-                        rh_surface_apply_warpfield,
-                        [("out_file", "warpfield")],
+                        rh_native_apply_warpfield,
+                        [("merged_file", "warpfield")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1089]:
```suggestion
                        [("out_file", "warpfield")],
```

File: xcp_d/workflow/anatomical.py [hunk 8]
```diff
@@ -953,134 +988,234 @@ def init_anatomical_wf(
...
+                        rh_32k_apply_affine,
+                        [("out_file", "affine")],
                     ),
-                    (rh_32k_midthick_wf, ds_midRsurf_wf, [("out_file", "in_file")]),
-                    (rh_32k_pial_wf, ds_pialRsurf_wf, [("out_file", "in_file")]),
-                    (rh_32k_wm_wf, ds_wmRsurf_wf, [("out_file", "in_file")]),
+                    (
+                        remerge_warpfield,
+                        rh_32k_apply_warpfield,
+                        [("merged_file", "forward_warp")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1114]:
```suggestion
                        [("out_file", "forward_warp")],
```

File: xcp_d/workflow/anatomical.py [hunk 9]
```diff
@@ -1184,16 +1319,41 @@ def init_anatomical_wf(
...
+                    ),
+                    (
+                        remerge_warpfield,
+                        lh_32k_hcpmidthick_apply_warpfield,
+                        [("merged_file", "forward_warp")],
+                    ),
+                    (
+                        remerge_inv_warpfield,
+                        lh_32k_hcpmidthick_apply_warpfield,
+                        [("merged_file", "warpfield")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1348]:
```suggestion
                        [("out_file", "warpfield")],
```

File: xcp_d/workflow/anatomical.py [hunk 10]
```diff
@@ -1204,16 +1364,41 @@ def init_anatomical_wf(
...
+                    ),
+                    (
+                        convert_xfm2world,
+                        rh_32k_hcpmidthick_apply_affine,
+                        [("out_file", "affine")],
+                    ),
+                    (
+                        remerge_warpfield,
+                        rh_32k_hcpmidthick_apply_warpfield,
+                        [("merged_file", "forward_warp")],
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1388]:
```suggestion
                        [("out_file", "forward_warp")],
```

File: xcp_d/workflow/anatomical.py [hunk 11]
```diff
@@ -684,72 +673,118 @@ def init_anatomical_wf(
...
-                name="lh_32k_pial_wf",
+                name="rh_32k_resample_wf",
                 mem_gb=mem_gb,
                 n_procs=omp_nthreads,
             )
+            rh_32k_resample_wf.iterables = (
+                "in_file",
+                [R_midthick_surf, R_pial_surf, R_wm_surf],
+            )
+
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:703]:
```suggestion

```
The linter flagged trailing whitespace.

File: xcp_d/workflow/anatomical.py [hunk 12]
```diff
@@ -953,134 +988,236 @@ def init_anatomical_wf(
...
                     (
                         surface_sphere_project_unproject_rh,
-                        rh_32k_midthick_wf,
-                        [("out_file", "current_sphere")],
-                    ), (
-                        surface_sphere_project_unproject_rh,
-                        rh_32k_pial_wf,
+                        rh_32k_resample_wf,
                         [("out_file", "current_sphere")],
+                    ),
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:1102]:
```suggestion
                    ),
```
The linter flagged trailing whitespace.

File: xcp_d/workflow/anatomical.py [hunk 13]
```diff
@@ -647,25 +648,13 @@ def init_anatomical_wf(
...
-                    template="fsaverage",
-                    hemi="L",
-                    density="164k",
-                    desc="std",
-                    suffix="sphere",
-                )
-            )
+            fs_L2fsLR = pkgrf('xcp_d',
+                              ('data/standard_mesh_atlases/fs_L/'
+                               'fs_L-to-fs_LR_fsaverage.L_LR.spherical_std.164k_fs_L.surf.gii'))
```
Comment (tsalo) [inline xcp_d/workflow/anatomical.py:653]: Why drop templateflow?

Comment (tsalo) [reply xcp_d/workflow/anatomical.py:653]: Answered in today's meeting- templateflow is missing one of the spheres that the DCAN lab wants. Do you know which one?

**Review by tsalo — APPROVED** [review]
I'd like to dramatically expand the docstring of the workflow function, but I can do that in a separate PR. The changes LGTM.

**Comment by tsalo** [comment]
One other question- what's the difference between `hcpmidthickness`, for example, and `midthickness`?

File: xcp_d/workflow/anatomical.py (hunk 13 above)
Comment (madisoth) [reply xcp_d/workflow/anatomical.py:653]: The sphere I couldn't find in TemplateFlow is the `fs_L-to-fs_LR_fsaverage.L_LR.spherical_std.164k_fs_L.surf.gii` (and the corresponding RH sphere). It's possible I didn't look hard enough for it though. I assumed the `tpl-fsaverage_hemi-L_den-164k_desc-std_sphere.surf.gii` was the same but was mistaken, hence this change.

**Comment by madisoth** [comment]
> One other question- what's the difference between `hcpmidthickness`, for example, and `midthickness`?

"midthickness" is derived from the FreeSurfer graymid (`mris_expand` with distance=0.5 applied to the WM surfs), while `hcpmidthickness` is a simple averaging of corresponding WM and pial vertices.

Basically the difference between (2) and (3) described in the 1st reply here: https://neurostars.org/t/midthickness-for-fsaverage/16676

File: xcp_d/workflow/anatomical.py (hunk 13 above)
Comment (tsalo) [reply xcp_d/workflow/anatomical.py:653]: Wow, that naming convention is... dense. Is it the transform of the fsaverage left hemisphere to fsLR at 164k density?

**Comment by tsalo** [comment]
Thanks for the clarification. Based on Chris's response, it _seems_ like the HCP midthickness file is objectively worse than the regular one, and should only be used if we don't want to spend the time/energy to run `mris_expand`. Why do we want it?

Also, do you know why we have HCP versions of inflated and veryinflated, but not the regular versions?

On an unrelated note, I'd like to continue both open threads in this PR, but I have other PRs that depend on your changes, so I'm going to merge the PR.

**Pull request merged**